internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -137,9 +137,9 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
               if (!ignore)
                 binaryDependency(zip.file.toPath, binaryClassName)
             }
-          case pf: xsbt.Compat.PlainNioFile =>
+          case pf: ZincCompat.PlainNioFile =>
```

Comment:
Before this PR, in 2.12 whenever the compiler gives Zinc a `scala.reflect.io.PlainNioFile` Zinc checks if it's a `xsbt.PlainNioFile`, which it will never be, and fails to register it as a dependency.

That's what this PR is fixing.